src/app/rpmostree-compose-builtin-tree.cxx

```diff
@@ -956,6 +951,23 @@ impl_install_tree (RpmOstreeTreeComposeContext *self,
   if (g_strcmp0 (g_getenv ("RPM_OSTREE_BREAK"), "post-yum") == 0)
     return FALSE;
 
+  *out_changed = TRUE;
+  return TRUE;
+}
+
+
+static gboolean
+impl_install_tree_post (RpmOstreeTreeComposeContext *self,
+                        GCancellable    *cancellable,
+                        GError         **error)
+{
+  /* Pick "next version" label from metadata */
+  gpointer vp = g_hash_table_lookup (self->metadata, OSTREE_COMMIT_META_KEY_VERSION);
+  auto v = static_cast<GVariant*>(vp);
+  g_assert (v != NULL);
+  g_assert (g_variant_is_of_type (v, G_VARIANT_TYPE_STRING));
+  auto next_version = rust::String(static_cast<char*>(g_variant_dup_string (v, NULL)));
```

Comment:
I believe this is leaking memory right now, the `char*` is being implicitly cast to `const char*` for the constructor and hence leaked.   It should just work instead to do `auto next_version = rust::String(g_variant_get_string (v, NULL));`.

Oh oh, I see we were leaking it before too.

Reply:
I didn't pay much attention when moving the code around, but I think you are right, so it's a good time to fix the leak too :)
I'll try your suggestion.